File: services/ordering/src/main/java/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/OrderProcessingError.java
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing;

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody;

import java.util.UUID;

public record OrderProcessingError(UUID orderId, int version, String message, String details) implements IntegrationMessageBody {
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/OrderProcessingError.kt
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody
import java.util.*

data class OrderProcessingError(val id: UUID, val deliveryVersion: Int, val message: String, val details: String) : IntegrationMessageBody {
    override fun orderId(): UUID = id
    override fun version(): Int = deliveryVersion
}
